From 72abe02ba150546898d73010c30a814731bb457e Mon Sep 17 00:00:00 2001
From: Devendra Tewari <devendra.tewari@gmail.com>
Date: Thu, 15 Jul 2021 13:30:46 -0300
Subject: [PATCH] Add flag to enable adbd at build time

Signed-off-by: Devendra Tewari <devendra.tewari@gmail.com>
---
 .../android-tools/android-tools_5.1.1.r37.bb                 | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/meta-oe/recipes-devtools/android-tools/android-tools_5.1.1.r37.bb b/meta-oe/recipes-devtools/android-tools/android-tools_5.1.1.r37.bb
index e8992346b..692753e2c 100644
--- a/meta-oe/recipes-devtools/android-tools/android-tools_5.1.1.r37.bb
+++ b/meta-oe/recipes-devtools/android-tools/android-tools_5.1.1.r37.bb
@@ -141,6 +141,10 @@ do_install() {
     if echo ${TOOLS} | grep -q "adbd" ; then
         install -d ${D}${bindir}
         install -m0755 ${B}/adbd/adbd ${D}${bindir}
+        if [ "${USB_DEBUGGING_ENABLED}" = "1" ] ; then
+           mkdir ${D}${localstatedir}
+           touch ${D}${localstatedir}/usb-debugging-enabled
+        fi
     fi
 
     # Outside the if statement to avoid errors during do_package
@@ -166,6 +170,7 @@ RDEPENDS_${PN}-fstools = "bash"
 FILES_${PN}-adbd = "\
     ${bindir}/adbd \
     ${systemd_unitdir}/system/android-tools-adbd.service \
+    ${@bb.utils.contains('USB_DEBUGGING_ENABLED', '1', '${localstatedir}/usb-debugging-enabled', '',d)} \
 "
 
 FILES_${PN}-fstools = "\
-- 
2.29.2

